### PR TITLE
Dockerfile: Change WORKDIR, remove redundant logic

### LINKED ISSUE #8456: Change WORKDIR away from /root
The current `Dockerfile` uses `WORKDIR=/root` which prevents running Cilium as non-root.

- [ ] Change the WORKDIR
- [ ] Build a container image with the change `DOCKER_IMAGE_TAG=test make dev-docker-image`
- [ ] Make any additional required changes which may depend on WORKDIR=/root

### PR DESCRIPTION
**Pull request by errordeveloper**
Fixes: #8456

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10531)

### CODE REVIEW

**Comment by errordeveloper** [comment]
test-me-please

**Comment by errordeveloper** [comment]
test-me-please

**Comment by errordeveloper** [comment]
So it looks like there is an implicit dependency on `Dockerfile`, and adding it to `.dockerignore` breaks that dependency. I should probably drop 0377ce2 from here and open an issue to investigate what's going on.

**Comment by errordeveloper** [comment]
test-me-please

File: Dockerfile [hunk 1]
```diff
@@ -44,10 +44,8 @@ COPY --from=cilium-envoy / /
 COPY plugins/cilium-cni/cni-install.sh /cni-install.sh
 COPY plugins/cilium-cni/cni-uninstall.sh /cni-uninstall.sh
 COPY contrib/packaging/docker/init-container.sh /init-container.sh
-WORKDIR /root
+WORKDIR /home/cilium
 RUN groupadd -f cilium \
-	&& echo ". /etc/profile.d/bash_completion.sh" >> /root/.bashrc \
```
Comment (nebril) [inline Dockerfile:49]: Won't this cause the image to not have any bash comletion at all? Maybe we should move bash completion to `/home/cilium/.bashrc`?

File: cilium-dev.Dockerfile [hunk 2]
```diff
@@ -27,9 +27,7 @@ RUN for i in proxylib envoy plugins/cilium-cni bpf cilium daemon cilium-health b
      do LOCKDEBUG=$LOCKDEBUG PKG_BUILD=1 V=$V LIBNETWORK_PLUGIN=$LIBNETWORK_PLUGIN \
             SKIP_DOCS=true DESTDIR= \
             make -C $i install; done
-RUN cd /root && groupadd -f cilium \
-	&& echo ". /etc/profile.d/bash_completion.sh" >> /root/.bashrc \
```
Comment (nebril) [inline cilium-dev.Dockerfile:31]: Same as above.

**Review by nebril — APPROVED** [review]

**Review by nebril — CHANGES_REQUESTED** [review]
Clicked "approve" by accident

File: Dockerfile (hunk 1 above)
Comment (errordeveloper) [reply Dockerfile:49]: See commit message in 9a65f5d8e4ecf54bf3449eccdd9593d595c79461.

I have tested it and `/etc/bash_completion.d/cilium` is already in place.

I did some more testing just now, and found that `/etc/bash_completion.d/cilium` gets loaded in a login shell, but doesn't get loaded in non-login shell.

So this works:
```
$ docker run -ti cilium bash -l
root@37e9f10b473c:/home/cilium# cilium 
bpf         completion  debuginfo   fqdn        kvstore     metrics     node        prefilter   service     version     
cleanup     config      endpoint    identity    map         monitor     policy      preflight   status      
root@37e9f10b473c:/home/cilium#
```
But this doesn't:
```
$ docker run -ti cilium bash   
root@f22ad80350d1:/home/cilium# cilium 
cilium                   cilium-bugtool           cilium-health            cilium-map-migrate       
cilium-agent             cilium-envoy             cilium-health-responder  cilium-node-monitor      
root@f22ad80350d1:/home/cilium# cilium 
```

I thin the fix should be simple, I'll have a look now.

Comment (errordeveloper) [reply Dockerfile:49]: By the way, the reason I wanted to get rid of these lines is to avoid references to `/root`.

File: Dockerfile [hunk 3]
```diff
@@ -44,10 +44,9 @@ COPY --from=cilium-envoy / /
...
-    && cilium completion bash >> /root/.bashrc \
+    && echo ". /etc/profile.d/bash_completion.sh" >> /etc/bash.bashrc \
```
Comment (errordeveloper) [inline Dockerfile:49]: See commit message in 341809fb9 for details.

I tested it like this:
```
$ docker run -ti cilium bash                                   
root@c12d417bc231:/home/cilium# cilium 
bpf         completion  debuginfo   fqdn        kvstore     metrics     node        prefilter   service     version     
cleanup     config      endpoint    identity    map         monitor     policy      preflight   status      
root@c12d417bc231:/home/cilium# exit
$ docker run -ti cilium bash -l
root@4bbd467f9a7f:/home/cilium# cilium 
bpf         completion  debuginfo   fqdn        kvstore     metrics     node        prefilter   service     version     
cleanup     config      endpoint    identity    map         monitor     policy      preflight   status      
root@4bbd467f9a7f:/home/cilium# logout
$
```

**Comment by errordeveloper** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by errordeveloper** [comment]
rebased

**Comment by errordeveloper** [comment]
test-me-please

**Comment by errordeveloper** [comment]
test-me-please

**Comment by errordeveloper** [comment]
I'm going to rebase now, looks like this is generally ready to go :)

**Comment by errordeveloper** [comment]
test-me-please

**Pull request merged**